File: src/main/java/com/codeperfection/shipit/dto/product/CreateProductDto.java
package com.codeperfection.shipit.dto.product;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateProductDto {

    @NotBlank
    private String name;

    @NotNull
    @Min(1)
    private Integer volume;

    @NotNull
    @Min(1)
    private Integer price;

    @NotNull
    @Min(0)
    private Integer countInStock;
}

File: src/main/kotlin/com/codeperfection/shipit/dto/product/CreateProductDto.kt
package com.codeperfection.shipit.dto.product

import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class CreateProductDto(
    @field:NotBlank
    @field:Size(max = 256)
    val name: String,
    @field:Min(1)
    val volume: Int,
    @field:Min(1)
    val price: Int,
    @field:Min(0)
    val countInStock: Int
)
